include/semver/semver.hpp

```diff
@@ -31,17 +31,29 @@ SOFTWARE.
 
 namespace semver
 {
+
+    typedef uint64_t BuildIdentifierPart;
+
     const std::string default_prerelease_part = "0";
     const char prerelease_delimiter = '.';
     const std::string numbers = "0123456789";
     const std::string prerelease_allowed_chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-";
-    const std::string version_pattern = "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$";
-    const std::string loose_version_pattern = "^v?(0|[1-9]\\d*)(?:\\.(0|[1-9]\\d*))?(?:\\.(0|[1-9]\\d*))?(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$";
+    const std::string version_pattern = "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)"
+                                        "(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))"
```

Comment:
@Twingate/sdwan, 
maybe you can help me understand this regex and if it complies with [RFC ABNF](https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions) .
```
<pre-release identifier> ::= <alphanumeric identifier>
                           | <numeric identifier>
```
 I would do something much easier on this one `(?:-[a-zA-Z0-9\.]*)`

Reply:
actually the regex it taken [from RFC](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string)